src/main/java/at/hannibal2/skyhanni/features/misc/FrogMaskFeatures.kt

```diff
@@ -0,0 +1,141 @@
+    private val frogMask by lazy { FROG_MASK.getItemStack() }
+
+    @HandleEvent(GuiRenderEvent.GuiOverlayRenderEvent::class, onlyOnIsland = IslandType.THE_PARK)
+    fun onRenderOverlay() {
+        if (!isEnabled()) return
+
+        config.displayPosition.renderRenderable(display, posLabel = "Frog Mask Display")
+    }
+
+    @HandleEvent(SecondPassedEvent::class, onlyOnIsland = IslandType.THE_PARK)
+    fun onSecondPassed() {
+        if (!isEnabled()) return
+        display = null
+
+        val helmet = InventoryUtils.getHelmet() ?: return
+        if (helmet.getInternalName() != FROG_MASK) return
+
+        activeRegionPattern.firstMatcher(helmet.getLore()) {
+            val currentRegion = group("region")
+
+            if (config.warning.enabled) lastWarning = handleWarning(currentRegion)
+            if (config.display) display = handleDisplay(currentRegion)
+        }
+    }
+
+    private fun handleWarning(currentRegion: String): SimpleTimeMark {
+        if (config.warning.warningType == WarningType.NEVER) return lastWarning
+
+        currentAreaPattern.firstMatcher(ScoreboardData.sidebarLinesFormatted) {
+            val needsToWarn =
+                group("area") != currentRegion && lastWarning.passedSince() > config.warning.cooldown.seconds
+
+            if (!needsToWarn) return lastWarning
+
+            when (config.warning.warningType) {
+                WarningType.BEING -> TitleManager.sendTitle("§cWrong Region!")
+                WarningType.FORAGING -> if (isForaging()) TitleManager.sendTitle("§cWrong Region!")
+                else -> return lastWarning
+            }
+
+            return SimpleTimeMark.now()
+        }
+
+        return lastWarning
+    }
+
+    private fun handleDisplay(currentRegion: String): Renderable {
+        val now = SkyBlockTime.now()
+        val timeRemaining = SkyBlockTime(year = now.year, month = now.month, day = now.day + 1).asTimeMark()
+        return updateDisplay(currentRegion, timeRemaining)
+    }
+
+    private fun updateDisplay(currentRegion: String, timeRemaining: SimpleTimeMark): Renderable {
+        val until = timeRemaining.timeUntil()
+        val timeString = until.format()
+
+        return Renderable.horizontalContainer(
+            listOf(
+                Renderable.itemStack(frogMask),
+                Renderable.string(
+                    "§5Frog Mask§6 - $currentRegion §6for §b$timeString",
+                ),
+            ),
+            spacing = 1,
+            verticalAlign = RenderUtils.VerticalAlignment.CENTER,
+        )
+    }
+
+    @HandleEvent
+    fun onConfigFix(event: ConfigUpdaterMigrator.ConfigFixEvent) {
+        event.move(74, "misc.frogMaskDisplay", "misc.frogMaskFeatures.frogMaskDisplay")
+        event.move(74, "misc.frogMaskDisplayPosition", "misc.frogMaskFeatures.frogMaskDisplayPosition")
+    }
+
+    private fun isForaging(): Boolean {
+        if (SkillApi.activeSkill != SkillType.FORAGING) return false
+        updateSkillInfo()
```

Comment:
instead of using the skill info thingy, rather write a "time since last log clicked at" thingy